### PR TITLE
Split extratest to several subcategories

### PR DESCRIPTION
**Pull request by asmorodskyi**
Implementing https://progress.opensuse.org/issues/16020.
1. Extratest will be splited in two main categories - UI and console tests
2. Yast related tests will not run with extratest they have another trigger
3. Helper functions used before only in opensuse main.pm now moved to main_common
to use them in both places.

### CODE REVIEW

**Comment by asmorodskyi** [comment]
http://kimball.arch.suse.de/tests/235 - prove run for console mode . UI and yast runs are on the way

**Comment by asmorodskyi** [comment]
http://kimball.arch.suse.de/tests/236 - DESKTOP=gnome

**Comment by asmorodskyi** [comment]
@dzedro might be interesting for you

**Comment by asmorodskyi** [comment]
it appears that yast ui tests currently not running in osd ( Y2UITEST is not used anywhere in sle main.pm ) so I can't break them with my commit

File: products/sle/main.pm [hunk 1]
```diff
@@ -1215,8 +1204,20 @@ elsif (get_var("Y2UITEST")) {
     # setup $serialdev permission and so on
```
Comment (coolo) [inline products/sle/main.pm]: I have Y2UITEST on screen on the diff - so your statement is confusing

Comment (asmorodskyi) [inline products/sle/main.pm]: @coolo you right , I wanted to say that I fail to find test suite which using this variable , so I suspecting that there is no job which run tests which are in this if branch. Sorry for confusion.

File: lib/main_common.pm [hunk 2]
```diff
@@ -86,6 +88,13 @@ sub setup_env {
     }
 }
 
+sub any_desktop_is_applicable() {
```
Comment (kravciak) [inline lib/main_common.pm]: Check if you are not duplicating is_desktop_installed from utils

File: lib/main_common.pm [hunk 3]
```diff
@@ -86,6 +88,13 @@ sub setup_env {
...
+    return get_var("DESKTOP") !~ /textmode/;
+}
+
+sub console_is_applicable() {
```
Comment (kravciak) [inline lib/main_common.pm]: I don't like naming here, console is applicable also when we have desktop

File: lib/main_common.pm (hunk 2 above)
Comment (asmorodskyi) [inline lib/main_common.pm]: is_desktop_installed - get_var("DESKTOP") !~ /textmode|minimalx/;

I actually also not very happy about both functions , their naming and whole idea behind. But I think that it's really bad idea in terms of one PR which suppose to solve problem A try to also problem B just because it touch same code. I didn't create this functions I just move them from one file to another.

Comment (kravciak) [inline lib/main_common.pm]: You are right, I started commenting before I read whole PR..

File: lib/main_common.pm (hunk 3 above)
Comment (asmorodskyi) [inline lib/main_common.pm]: 1. see answer to previous comment 
2. also discuss this with @okurz  , so probably he can add his 5 cents here . Oli what do you think should we rename it ? 
And again even if we need to rename it I think it should be different PR to not create a mess in git log

File: products/sle/main.pm (hunk 1 above)
Comment (okurz) [inline products/sle/main.pm]: now as you split it the same way as used for o3 you would need to also sync the test suites, i.e. define a yast ui test suite.

File: lib/main_common.pm (hunk 3 above)
Comment (okurz) [inline lib/main_common.pm]: yes, propose renaming but outside this PR. What you do is extract an existing method into a common package that is included from multiple main.pm's. That makes sense and keep it for now.

File: products/sle/main.pm (hunk 1 above)
Comment (asmorodskyi) [inline products/sle/main.pm]: @okurz sure thing but because they never exists on o3 I can't introduce regression so I don't need prove run :) I will do this AFTER PR will be merged

**Comment by sysrich** [comment]
+0.5 from me - I like the direction of travel of this PR, but the confusion with naming is certainly something that will have to be tidied up if this PR is accepted.

I'm not comfortable merging this until we can be sure the YaST extratests will run on osd. They are some of the most important extratests for osd.

**Comment by okurz** [comment]
In the current state there is no test suite on osd specifying Y2UITEST. That means the YaST GUI tests are *not run at all* on osd anyway and also most of the ncurses YaST UI tests are not. What @asmorodskyi did so far is harmonize the structure of SLE and openSUSE tests more so that we can actually see where they really differ. This PR is already providing a benefit without changing a lot of the test plan (only minor shifts which have been proven to work in the prove runs). I will merge this PR. @asmorodskyi will improve the code further on and should do the following:
* check YaST GUI tests on SLE
* extract common functions into e.g. `lib/main_common.pm`
* check remaining differences if they make sense

File: products/opensuse/main.pm [hunk 4]
```diff
@@ -515,6 +507,7 @@ sub load_extra_tests() {
         if (get_var("FILESYSTEM", "btrfs") eq "btrfs") {
             loadtest "console/btrfs_autocompletion";
             if (get_var("NUMDISKS", 0) > 1) {
+                loadtest "console/snapper_cleanup";
```
Comment (DimStar77) [inline products/opensuse/main.pm]: https://openqa.opensuse.org/tests/340474 -> That part seems not to work yet.. Product bug? openQA issue?

**Comment by DimStar77** [comment]
@okurz https://openqa.opensuse.org/tests/340694#settings sets Y2UITEST

File: products/opensuse/main.pm (hunk 4 above)
Comment (okurz) [inline products/opensuse/main.pm]: https://openqa.opensuse.org/tests/340474#step/snapper_cleanup/1 shows that there is no "root" config for snapper. That probably means that the image created is so small that the installer correctly suggests to not create a filesystem with snapshots.

@asmorodskyi did you or can you crosscheck the HDD image sizes on osd/o3 regarding this?

**Comment by okurz** [comment]
that's the reason why we wanted to restructure the tests in the first place, to see the differences, make sure no module is expected to be run but is not actually and such
